### PR TITLE
[#157052270], [#156990275] User can track asset assignment, More details on Asset list page

### PR DESCRIPTION
**Pull request by HawiCaesar**
### What does this PR do ?
- Allows a system user to track asset assignment
- Allow the Admin to see more details on the Asset list page 
### Description of work done
- Create `AllocationsHistory` model 
- Add `asset`, `previous_owner`, `current_owner`,  `created_at` fields to model
- Add `post_save` method that makes a `current_owner` the `previous_owner` when reassignment is done
- write tests
- add `list_display` in `AssetAdmin` to show more details of an asset

### Screenshot
screen shot 2018-04-27 at 10 23 21

### Relevant PT story
[#157052270](https://www.pivotaltracker.com/story/show/157052270)
[#156990275](https://www.pivotaltracker.com/story/show/156990275)

### CODE REVIEW

File: api/migrations/0030_auto_20180426_1313.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            ],
+        ),
+        migrations.RemoveField(
+            model_name='asset',
+            name='assigned_to',
+        ),
+        migrations.AddField(
+            model_name='asset',
+            name='current_owner',
+            field=models.CharField(default=django.utils.timezone.now, editable=False, max_length=50),
```
Comment (egichuri) [inline api/migrations/0030_auto_20180426_1313.py]: please change this default.. date isn't valid for an integer field

File: api/models.py [hunk 2]
```diff
@@ -308,3 +345,10 @@ def save_initial_asset_status(sender, **kwargs):
         AssetStatus.objects.create(asset=current_asset,
                                    current_status="Available")
         current_asset.save()
+
+
+@receiver(post_save, sender=AllocationHistory)
+def save_current_asset_owner(sender, **kwargs):
+    asset_owner = kwargs.get('instance')
+    asset_owner.asset.current_owner = asset_owner.current_owner
```
Comment (egichuri) [inline api/models.py]: `asset_owner.asset.assigned_to`

File: api/models.py [hunk 3]
```diff
@@ -292,6 +293,42 @@ def save(self, *args, **kwargs):
...
+    previous_owner = models.ForeignKey('User',
+                                       related_name='previous_owner_asset',
+                                       editable=False,
+                                       blank=True,
+                                       null=True,
+                                       on_delete=models.PROTECT)
+    created_at = models.DateTimeField(auto_now_add=True, editable=False)
+
+    class Meta:
+        verbose_name_plural = "Allocation Histories"
```
Comment (egichuri) [inline api/models.py]: "Allocation History" works better here

File: api/migrations/0030_auto_20180426_1313.py (hunk 1 above)
Comment (HawiCaesar) [reply api/migrations/0030_auto_20180426_1313.py]: Done.

File: api/models.py (hunk 3 above)
Comment (HawiCaesar) [reply api/models.py]: I guess so but`histories` doesn't sound so bad does it ?
screen shot 2018-04-30 at 12 57 02

File: api/models.py (hunk 2 above)
Comment (HawiCaesar) [reply api/models.py]: We have changed this.

File: api/models.py (hunk 3 above)
Comment (egichuri) [reply api/models.py]: it's like app changelog vs changelogs.. :)

**Pull request merged**